### PR TITLE
Add public blog

### PR DESCRIPTION
**Pull request by Shreyanand**
## Related Issues and Dependencies
https://github.com/operate-first/operate-first.github.io/issues/37
https://github.com/operate-first/operate-first.github.io/issues/31

## This introduces a breaking change

- [ ] Yes
- [X] No

## This Pull Request implements

content manipulation for the operate first website.

## Description

- Add the material for public categorical encoding blog.
- Fix reference rendering by escaping special characters

### CODE REVIEW

**Review by oindrillac — APPROVED** [review]
Looks good to me :rocket:

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
  - For example, all _Civil Engineer I_ will be represented by integer _k_
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
  - E.g. Categories: _Associate Software Developer_, _Software Developer_, _Senior Software Developer_
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
  - If we wanted to use word embedding methods such as word2vec, we would have to go through a cleaning phase: those algorithms are not trained to work on data such as _Accountant/Auditor I_. However, this can be error-prone and time-consuming.
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
  - The high dimensions can be meaningfully reduced to lower dimensions: _All_ to _Software developer_s, similar to clustering.
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
  - Similar to [LDA](https://en.wikipedia.org/wiki/Latent_Dirichlet_allocation) for finding topics in documents but since the string entries here are much shorter than text documents and can contain typos, it uses ngrams level representation.
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
- **Outlier detection:** To find the outliers, we use the [isolation forest](https://en.wikipedia.org/wiki/Isolation_forest) algorithm that finds points that are easiest to isolate.
```

or 
```suggestion
- **Outlier detection:** To find the outliers, we use the [isolation forest](https://cs.nju.edu.cn/zhouzh/zhouzh.files/publication/tkdd11.pdf) algorithm that finds points that are easiest to isolate.
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
- One-hot encoded variable: if a node says that `vendor_seabios <= 0.5`, that means that for all the systems where the vendor is not _seabios_. Since one-hot encoding means 0 or 1, &lt; 0.5 means 0.
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
- Gammas Poisson factorized variables: if a node says `modelname_broadwell,skylake,haswell <= R` (real number), then that means that systems that have strings _broadwell_, _skylake_ or _haswell_ in their _modelname_ are important factors for splitting the tree further. In order to decide whether their presence or absence causes these anomalies, further validation has to be done.
```

File: docs/blog/blog.md
Comment (tumido) [inline docs/blog/blog.md]:
```suggestion
- Example rule path found in Figure 4: if (modelname has _platinum_, _pentium_, _8175m_) and (vendor is _seabios_) then it could be an anomaly.
```

**Review by tumido — COMMENTED** [review]
LGTM! I just have couple of suggestions to improve readability:

1) Nitpick: consistency in formatting terms
2) Adding a source to LDA and isolation forest

**Comment by Shreyanand** [comment]
> LGTM! I just have couple of suggestions to improve readability:
> 
> 1. Nitpick: consistency in formatting terms
> 2. Adding a source to LDA and isolation forest

Thanks for the suggestions @tumido I added them all to the blog.

For improving my git skills, I'm writing what I did in this PR. 
1) Use UI to add all the readability suggestions to a batch commit
2) Pull the PR locally and make other changes in a temp branch
3) Merge temp branch to master branch
4) Squash commits and force push to my fork

If there is a better workflow, I'd like to know.

**Review by tumido — APPROVED** [review]

**Comment by tumido** [comment]
Sorry to keep you waiting on this @Shreyanand I've just merged it.

About your workflow. I think it's good overall, but it can be simplified:

1. Unfortunately there's on other way to accept the GitHub suggestions then via UI.
2. There's nothing wrong in leaving the "batch" commit with the suggestions in the branch, no need to squash it. If you need to squash it though for some reason, you did it the right way. For other squashing options see point 4.
3. I would suggest you to use feature (topic) branches instead of master when working on PRs. It's better to keep the `master` branch always in sync with `upstream/master` and branch from it for each feature you're working on. This way you can work on multiple branches at the same time and once your PR is merged you just pull to your `master` from `upstream/master` (you update it). It also allows you to update the branch while the PR is open. So, if you'd accept a suggestion in a PR opened from branch `x` you just pull the changes on your local branch `x` from your origin. Or if you want to push a new commit to the PR, you just commit it on your local `x` branch and push it to origin. All of this keeps your `master` unaffected and ready to be updated with merged changes from others, or letting you start work on a different PR from the same commit. See:
    - https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow
    - https://guides.github.com/introduction/flow/
    - https://git-scm.com/book/en/v2/Git-Branching-Branching-Workflows
4. You can ask for commits to be squashed on PR merge, instead of doing it manually. It may be preferred flow on some projects - especially the big ones with high volume of commits. I personally favor the rebase and merge or merge commits on small projects like these, since I like the granularity. Further read on this:
    - https://rietta.com/blog/github-merge-types/
    - https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/about-pull-request-merges

**Pull request merged**
